### PR TITLE
[FIX] minor optimization

### PR DESCRIPTION
**Pull request by timosachsenberg**
Should result in a significant speed up of the filtering step

### CODE REVIEW

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp [hunk 1]
```diff
@@ -93,9 +95,11 @@ namespace OpenMS
         mz_shifts_actual.push_back(std::numeric_limits<double>::quiet_NaN());
         mz_shifts_actual_indices.push_back(-1);
       }
-
     }
 
+    // early-out: not peaks found 
+    if (found_peaks < peaks_per_peptide_min_) return -1;
```
Comment (lars20070) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp]: `peaks_per_peptide_min_ * pattern.getMassShiftsCount()`

`found_peaks_` counts all m/z shifts. `peaks_per_peptide_min_` refers to a single peptide. `pattern.getMassShiftsCount()` is the number of peptides, e.g. 3 for a triplet search.

Comment (timosachsenberg) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp]: I see so I can savely early out if (found_peaks < peaks_per_peptide_min_ * pattern.getMassShiftsCount())

Comment (lars20070) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp]: Yes. I further speedup.

And the entire check can now be done within the method and not here
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.cpp#L199
Same here
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringCentroided.cpp#L164

**Comment by timosachsenberg** [comment]
ok made the changes

**Comment by timosachsenberg** [comment]
any more comments?

**Comment by lars20070** [comment]
The early-out is still unchanged.
https://github.com/OpenMS/OpenMS/pull/1622/files#diff-d07005aa96a93e57990ba6d481c8425eR101

**Comment by timosachsenberg** [comment]
strange... I readd it? otherwise fine in your view?

**Comment by lars20070** [comment]
In https://github.com/timosachsenberg/OpenMS/commit/c3ab78aa4eb1bdbe311948bb8948cc8ac1b690a3 only the comment is changed.

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringCentroided.cpp [hunk 2]
```diff
@@ -160,6 +160,10 @@ namespace OpenMS
            */
           vector<double> mz_shifts_actual; // actual m/z shifts (differ slightly from expected m/z shifts)
           vector<int> mz_shifts_actual_indices; // peak indices in the spectrum corresponding to the actual m/z shifts
+
+          mz_shifts_actual_indices.reserve(patterns_[pattern].getMZShiftCount());
```
Comment (lars20070) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringCentroided.cpp]: Equivalent changes here as well? https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.cpp#L197

**Comment by lars20070** [comment]
+1

Very nice changes in `getPeakIndex()`. Thanks!

**Pull request merged**